### ISSUE TITLE
ChainActivityMessage infinite queue issue

### ISSUE THREAD

**Issue by e-five256** [MbinOrg/mbin#668]
**Describe the bug**

In some situations, a `ChainActivityMessage` may come in that loops infinitely, causing queue processing to stall

**On which Mbin instance did you find the bug?**
fedia.io

**Which Mbin version was running on the instance?**
1.5.2

**Comment by e-five256** [comment]
```json
{
  "chain": [
    {
      "@context": [
        "https:\\/\\/join-lemmy.org\\/context.json",
        "https:\\/\\/www.w3.org\\/ns\\/activitystreams"
      ],
      "type": "Note",
      "id": "https:\\/\\/lemmy.one\\/comment\\/2894730",
      "attributedTo": "https:\\/\\/lemmy.one\\/u\\/lemann",
      "to": [
        "https:\\/\\/www.w3.org\\/ns\\/activitystreams#Public"
      ],
      "cc": [
        "https:\\/\\/lemmy.world\\/c\\/technology",
        "https:\\/\\/kbin.social\\/u\\/geosoco"
      ],
      "content": "<p>That\\u2019s going to hurt any kind of confidence users have for Microsoft-designed mobile devices going forward IMO.<\\/p>\n<p>A bit of a shame since the Duo was a pretty nice looking device<\\/p>\n",
      "inReplyTo": "https:\\/\\/kbin.social\\/m\\/technology@lemmy.world\\/t\\/445850",
      "mediaType": "text\\/html",
      "source": {
        "content": "That's going to hurt any kind of confidence users have for Microsoft-designed mobile devices going forward IMO.\n\nA bit of a shame since the Duo was a pretty nice looking device",
        "mediaType": "text\\/markdown"
      },
      "published": "2023-09-11T18:22:20.526313Z",
      "tag": [
        {
          "href": "https:\\/\\/kbin.social\\/u\\/geosoco",
          "name": "@geosoco@kbin.social",
          "type": "Mention"
        }
      ],
      "distinguished": false,
      "language": {
        "identifier": "en",
        "name": "English"
      },
      "audience": "https:\\/\\/lemmy.world\\/c\\/technology"
    }
  ],
  "parent": null,
  "announce": null,
  "like": {
    "id": "https:\\/\\/lemmy.blahaj.zone\\/activities\\/like\\/1d5a1088-be4a-413c-9e89-5cbac2ee4368",
    "actor": "https:\\/\\/lemmy.blahaj.zone\\/u\\/mraow_",
    "@context": [
      "https:\\/\\/join-lemmy.org\\/context.json",
      "https:\\/\\/www.w3.org\\/ns\\/activitystreams"
    ],
    "object": "https:\\/\\/literature.cafe\\/comment\\/1157238",
    "type": "Like",
    "audience": "https:\\/\\/lemmy.ml\\/c\\/fediverse"
  },
  "dislike": null
}
```

**Comment by e-five256** [comment]
Would like help dissecting this if possible.

It reads like: a `lemmy.blahaj.zone` user liked a `lemmy.one` [post](https://lemmy.one/comment/2894730) in an entry by a `kbin.social` user on a `lemmy.world` magazine. If that is all true, the part I'm struggling to understand is what `lemmy.ml/c/fediverse` or `literature.cafe` have to do with any of that

**Comment by BentiGorlich** [comment]
I just had one too:
```json
{
    "chain": [
        {
            "@context": [
                "https:\/\/www.w3.org\/ns\/activitystreams",
                {
                    "Hashtag": "as:Hashtag"
                }
            ],
            "id": "https:\/\/historywalksvenice.com\/venetian-story\/venice-access-fee-money-for-nothing\/",
            "type": "Note",
            "attachment": [
                {
                    "type": "Image",
                    "url": "https:\/\/historywalksvenice.com\/upload\/2023\/04\/Crowd-at-San-Marco.jpg",
                    "mediaType": "image\/jpeg",
                    "name": "Why I normally avoid Piazza San Marco if I can in any way"
                }
            ],
            "attributedTo": "https:\/\/historywalksvenice.com\/author\/info\/",
            "content": "<p><strong>Venice Access Fee &#8212; Money for nothing<\/strong><\/p><p>The &#8220;Venice Access Fee&#8221; introduced by the municipal administration will do absolutely nothing to resolve the real issues of the city.<\/p><p><a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/topics\/venice-today\/\">#VeniceToday<\/a> <a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/tag\/mass-tourism\/\">#MassTourism<\/a> <a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/tag\/venezia\/\">#Venezia<\/a> <a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/tag\/venice\/\">#Venice<\/a><\/p><p><a href=\"https:\/\/historywalksvenice.com\/venetian-story\/venice-access-fee-money-for-nothing\/\">https:\/\/historywalksvenice.com\/venetian-story\/venice-access-fee-money-for-nothing\/<\/a><\/p>",
            "contentMap": {
                "en": "<p><strong>Venice Access Fee &#8212; Money for nothing<\/strong><\/p><p>The &#8220;Venice Access Fee&#8221; introduced by the municipal administration will do absolutely nothing to resolve the real issues of the city.<\/p><p><a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/topics\/venice-today\/\">#VeniceToday<\/a> <a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/tag\/mass-tourism\/\">#MassTourism<\/a> <a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/tag\/venezia\/\">#Venezia<\/a> <a rel=\"tag\" class=\"hashtag u-tag u-category\" href=\"https:\/\/historywalksvenice.com\/tag\/venice\/\">#Venice<\/a><\/p><p><a href=\"https:\/\/historywalksvenice.com\/venetian-story\/venice-access-fee-money-for-nothing\/\">https:\/\/historywalksvenice.com\/venetian-story\/venice-access-fee-money-for-nothing\/<\/a><\/p>"
            },
            "published": "2024-02-04T05:53:15Z",
            "tag": [
                {
                    "type": "Hashtag",
                    "href": "https:\/\/historywalksvenice.com\/tag\/mass-tourism\/",
                    "name": "#MassTourism"
                },
                {
                    "type": "Hashtag",
                    "href": "https:\/\/historywalksvenice.com\/tag\/venezia\/",
                    "name": "#Venezia"
                },
                {
                    "type": "Hashtag",
                    "href": "https:\/\/historywalksvenice.com\/tag\/venice\/",
                    "name": "#Venice"
                },
                {
                    "type": "Mention",
                    "href": "https:\/\/mastodon.social\/users\/seindal",
                    "name": "@seindal@mastodon.social"
                },
                {
                    "type": "Mention",
                    "href": "https:\/\/a.gup.pe\/u\/venice",
                    "name": "@venice@a.gup.pe"
                },
                {
                    "type": "Mention",
                    "href": "https:\/\/a.gup.pe\/u\/histodons",
                    "name": "@histodons@a.gup.pe"
                }
            ],
            "updated": "2024-02-19T10:00:45Z",
            "url": "https:\/\/historywalksvenice.com\/venetian-story\/venice-access-fee-money-for-nothing\/",
            "to": [
                "https:\/\/www.w3.org\/ns\/activitystreams#Public",
                "https:\/\/historywalksvenice.com\/wp-json\/activitypub\/1.0\/users\/2\/followers"
            ],
            "cc": [
                "https:\/\/mastodon.social\/users\/seindal",
                "https:\/\/a.gup.pe\/u\/venice",
                "https:\/\/a.gup.pe\/u\/histodons"
            ]
        }
    ],
    "parent": null,
    "announce": {
        "@context": [
            "https:\/\/www.w3.org\/ns\/activitystreams",
            "https:\/\/w3id.org\/security\/v1",
            {
                "@context": {
                    "PropertyValue": "http:\/\/schema.org\/#PropertyValue"
                }
            }
        ],
        "id": "https:\/\/a.gup.pe\/s\/6610df1a18792c993b48933c",
        "type": "Announce",
        "actor": "https:\/\/a.gup.pe\/u\/histodons",
        "cc": "https:\/\/historywalksvenice.com\/author\/info\/",
        "likes": {
            "id": "https:\/\/a.gup.pe\/s\/6610df1a18792c993b48933c\/likes",
            "type": "OrderedCollection",
            "first": "https:\/\/a.gup.pe\/s\/6610df1a18792c993b48933c\/likes?page=true",
            "totalItems": 0
        },
        "object": "https:\/\/historywalksvenice.com\/social-media\/venice-access-fee-money-for-nothing\/",
        "published": "2024-04-06T05:35:22.855Z",
        "shares": {
            "id": "https:\/\/a.gup.pe\/s\/6610df1a18792c993b48933c\/shares",
            "type": "OrderedCollection",
            "first": "https:\/\/a.gup.pe\/s\/6610df1a18792c993b48933c\/shares?page=true",
            "totalItems": 0
        },
        "to": [
            "https:\/\/a.gup.pe\/u\/histodons\/followers",
            "as:Public"
        ]
    },
    "like": null,
    "dislike": null
}
```

**Comment by BentiGorlich** [comment]
I think I have the solution: the problem is purely based on we fetching URL A and get an object that has a different url as it's id B. So the ChainActivityHandler constantly creates an object with the apId B, but the Like-, Announce- and DislikeHandler are looking for an object with apId A, because that is what was referenced in the initial payload. 

In the first example we fetch the url `https://literature.cafe/comment/1157238`, but get an object with the id `https://lemmy.one/comment/2894730`. (I could not confirm this one by curling the URL, but I could reproduce it with my dev environment. Maybe they want http signatures? I don't know)

In the second example we fetch the url `https://historywalksvenice.com/social-media/venice-access-fee-money-for-nothing/`, but get an object with the id `https://historywalksvenice.com/venetian-story/venice-access-fee-money-for-nothing/`.
